### ISSUE TITLE
Screenreader reads command link targets

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode#211168]
Testing #210661

- Send a message with `@workspace`
- The screenreader spends a really long time reading the command link targets. It should probably only read the link text (the file name)

**Comment by meganrogge** [comment]
do you mean as a part of the response? can you clarify what you mean by "command link targets" @roblourens ?

**Comment by roblourens** [comment]
When you see this

<img width="395" alt="image" src="https://github.com/microsoft/vscode/assets/323878/69281794-92be-41a3-b985-0e07a9dc299f">

The markdown looks like this

```
The [`Button`](command:_github.copilot.openSymbolFromReferences?%5B%7B%22%24mid%22%3A1%2C%22fsPath%22%3A%22%2FUsers%2Froblou%2Fcode%2Fvscode%2Fsrc%2Fvs%2Fbase%2Fbrowser%2Fui%2Fbutton%2Fbutton.ts%22%2C%22external%22%3A%22file%3A%2F%2F%2FUsers%2Froblou%2Fcode%2Fvscode%2Fsrc%2Fvs%2Fbase%2Fbrowser%2Fui%2Fbutton%2Fbutton.ts%22%2C%22path%22%3A%22%2FUsers%2Froblou%2Fcode%2Fvscode%2Fsrc%2Fvs%2Fbase%2Fbrowser%2Fui%2Fbutton%2Fbutton.ts%22%2C%22scheme%22%3A%22file%22%7D%2C%7B%22line%22%3A74%2C%22character%22%3A0%7D%5D "src/vs/base/browser/ui/button/button.ts") class is defined in the file [`src/vs/base/browser/ui/button/button.ts`](command:_github.copilot.openRelativePath?%5B%7B%22scheme%22%3A%22file%22%2C%22authority%22%3A%22%22%2C%22path%22%3A%22%2FUsers%2Froblou%2Fcode%2Fvscode%2Fsrc%2Fvs%2Fbase%2Fbrowser%2Fui%2Fbutton%2Fbutton.ts%22%2C%22query%22%3A%22%22%2C%22fragment%22%3A%22%22%7D%5D "/Users/roblou/code/vscode/src/vs/base/browser/ui/button/button.ts"). This class provides the functionality for creating and managing buttons in the user interface.
```

and the screenreader spends a long time reading all the long command link targets. So somewhere we would need to present the screenreader with a version of this that has the links removed or replaced with what you want the screenreader user to hear.

**Comment by roblourens** [comment]
Although in the future, I would like to have an API that extensions can use to stream symbol references (we have this for file references, but vscode-copilot has not adopted it yet). That would make this case easier to fix. Command links could still exist from other extensions but might not show up very often.

**Comment by meganrogge** [comment]
Yeah this is a similar problem to https://github.com/microsoft/vscode/issues/210665 

@mjbvz what would you suggest here?

**Comment by meganrogge** [comment]
The screen reader will read any text that's present on the document. We cannot change that. So our only choice is to not provide such lengthy text. I don't know how we could do that and still have the link feature.

**Comment by mjbvz** [comment]
We already set a `title` attribute on the link. I'm surprised screen readers aren't using that for reading by default, while perhaps still allowing the user to hear the full `src` if requested 

Any idea @meganrogge what we would need to set to get the right behavior?

**Comment by meganrogge** [comment]
I believe this is working just fine in rendered markdown. It indicates the title first and then it says the source of that link. If screen reader users do not want to hear that, they will press Escape. 

https://github.com/microsoft/vscode/assets/29464607/7eccab01-d3e0-4368-9d0c-417bbfc7b4a4

**Comment by roblourens** [comment]
I was hearing this as it reads the full text of the response, not when the link is focused. I think in that case it's just reading the markdown of the response.

**Comment by meganrogge** [comment]
Oh, I haven't heard users complain about this, so it's probably fine. I think they likely press escape to quiet the noise of the response and investigate it once it's come in either in the accessible view or via tabbing.
